dali/python/nvidia/dali/experimental/eager.py

```diff
@@ -101,6 +119,57 @@ def __call__(self, inputs, kwargs):
     return EagerOperator
 
 
+def _iterator_op_factory(op_class, op_name, num_inputs, call_args_names):
+    class EagerOperator(_eager_op_base_factory(op_class, op_name, num_inputs, call_args_names)):
+        def __init__(self, call_args, *, max_batch_size, **kwargs):
+            pad_last_batch = kwargs.get('pad_last_batch', False)
+            kwargs['pad_last_batch'] = True
+
+            super().__init__(max_batch_size=max_batch_size, **kwargs)
+
+            self._call_args = call_args
+            self._iter = 0
+
+            epoch_size = self._backend_op.reader_meta()['epoch_size']
+            self._num_iters = math.ceil(epoch_size / max_batch_size)
+
+            # Size of the last batch in an epoch.
+            if pad_last_batch or epoch_size % max_batch_size == 0:
+                self._last_batch_size = max_batch_size
+            else:
+                self._last_batch_size = epoch_size % max_batch_size
+
+            assert isinstance(self._last_batch_size, int)
+
+        def __next__(self):
+            """ Iterates over dataset once per epoch (last batch may not be full). """
+
+            if self._iter < self._num_iters:
+                self._iter += 1
+                outputs = self._backend_op([], self._call_args)
+
+                if self._iter == self._num_iters:
+                    # Return potentially partial batch at the end of an epoch.
+                    outputs = [type(tl_output)(
+                        [tl_output[i] for i in range(self._last_batch_size)]) for tl_output in outputs]
+
+                if len(outputs) == 1:
+                    outputs = outputs[0]
+
+                return outputs
+            else:
```

Comment:
nitpick: I'd swap the if else. Maybe it is a matter of personal preference but when one of the branches is short code that interrupts the flow it reads better as a first branch.